File: lib/Models/UpdateCampaignOfferDTO.cpp


#include "UpdateCampaignOfferDTO.h"

using namespace Tiny;

UpdateCampaignOfferDTO::UpdateCampaignOfferDTO()
{
	offerId = std::string();
	quantum = QuantumDTO();
	available = bool(false);
	vat = int(0);
}

UpdateCampaignOfferDTO::UpdateCampaignOfferDTO(std::string jsonString)
{
	this->fromJson(jsonString);
}

UpdateCampaignOfferDTO::~UpdateCampaignOfferDTO()
{

}

void
UpdateCampaignOfferDTO::fromJson(std::string jsonObj)
{
    bourne::json object = bourne::json::parse(jsonObj);

    const char *offerIdKey = "offerId";

    if(object.has_key(offerIdKey))
    {
        bourne::json value = object[offerIdKey];



        jsonToValue(&offerId, value, "std::string");


    }

    const char *quantumKey = "quantum";

    if(object.has_key(quantumKey))
    {
        bourne::json value = object[quantumKey];




        QuantumDTO* obj = &quantum;
		obj->fromJson(value.dump());

    }

    const char *availableKey = "available";

    if(object.has_key(availableKey))
    {
        bourne::json value = object[availableKey];



        jsonToValue(&available, value, "bool");


    }

    const char *vatKey = "vat";

    if(object.has_key(vatKey))
    {
        bourne::json value = object[vatKey];



        jsonToValue(&vat, value, "int");


    }


}

bourne::json
UpdateCampaignOfferDTO::toJson()
{
    bourne::json object = bourne::json::object();





    object["offerId"] = getOfferId();







	object["quantum"] = getQuantum().toJson();





    object["available"] = isAvailable();






    object["vat"] = getVat();



    return object;

}

std::string
UpdateCampaignOfferDTO::getOfferId()
{
	return offerId;
}

void
UpdateCampaignOfferDTO::setOfferId(std::string  offerId)
{
	this->offerId = offerId;
}

QuantumDTO
UpdateCampaignOfferDTO::getQuantum()
{
	return quantum;
}

void
UpdateCampaignOfferDTO::setQuantum(QuantumDTO  quantum)
{
	this->quantum = quantum;
}

bool
UpdateCampaignOfferDTO::isAvailable()
{
	return available;
}

void
UpdateCampaignOfferDTO::setAvailable(bool  available)
{
	this->available = available;
}

int
UpdateCampaignOfferDTO::getVat()
{
	return vat;
}

void
UpdateCampaignOfferDTO::setVat(int  vat)
{
	this->vat = vat;
}

File: src/main/java/org/openapitools/server/api/model/UpdateCampaignOfferDTO.java
package org.openapitools.server.api.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import org.openapitools.server.api.model.QuantumDTO;

/**
 * Параметры размещения товара в магазине.
 **/
@JsonInclude(JsonInclude.Include.NON_NULL)
public class UpdateCampaignOfferDTO   {
  
  private String offerId;
  private QuantumDTO quantum;
  private Boolean available;
  private Integer vat;

  public UpdateCampaignOfferDTO () {

  }

  public UpdateCampaignOfferDTO (String offerId, QuantumDTO quantum, Boolean available, Integer vat) {
    this.offerId = offerId;
    this.quantum = quantum;
    this.available = available;
    this.vat = vat;
  }

    
  @JsonProperty("offerId")
  public String getOfferId() {
    return offerId;
  }
  public void setOfferId(String offerId) {
    this.offerId = offerId;
  }

    
  @JsonProperty("quantum")
  public QuantumDTO getQuantum() {
    return quantum;
  }
  public void setQuantum(QuantumDTO quantum) {
    this.quantum = quantum;
  }

    
  @JsonProperty("available")
  public Boolean getAvailable() {
    return available;
  }
  public void setAvailable(Boolean available) {
    this.available = available;
  }

    
  @JsonProperty("vat")
  public Integer getVat() {
    return vat;
  }
  public void setVat(Integer vat) {
    this.vat = vat;
  }


  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    UpdateCampaignOfferDTO updateCampaignOfferDTO = (UpdateCampaignOfferDTO) o;
    return Objects.equals(offerId, updateCampaignOfferDTO.offerId) &&
        Objects.equals(quantum, updateCampaignOfferDTO.quantum) &&
        Objects.equals(available, updateCampaignOfferDTO.available) &&
        Objects.equals(vat, updateCampaignOfferDTO.vat);
  }

  @Override
  public int hashCode() {
    return Objects.hash(offerId, quantum, available, vat);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class UpdateCampaignOfferDTO {\n");
    
    sb.append("    offerId: ").append(toIndentedString(offerId)).append("\n");
    sb.append("    quantum: ").append(toIndentedString(quantum)).append("\n");
    sb.append("    available: ").append(toIndentedString(available)).append("\n");
    sb.append("    vat: ").append(toIndentedString(vat)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}
